### ISSUE TITLE
Thoughts about Opensearch support?

### ISSUE THREAD

**Issue by jdt-at-litera** [mpenet/spandex#67]
I've been using spandex for some months researching/prototyping code based on Opensearch, which is a fork of Elasticsearch 7.10 driven by amazon.  We use it because we're not enamored of current Elasticsearch licensing practices.

Meanwhile, spandex was doing great.  However as I move it into production code and tests with orchestra.spec instrumentation enabled, I'm getting test failures because of trivial differences in Opensearch responses vs Elasticsearch. For example, Opensearch returns a :hosts key in the response, spandex is expecting a :host key.

I suppose Opensearch will diverge further as it's being actively changed and isn't targeted at 100% 7.x Elasticsearch compatibility forever.

Just looking for thoughts, info on existing forks already done for this purpose if they exist, or other suggestions.  Of course I could cease instrumenting spandex,  inject alternate specs for spandex where they are incompatible, or fork.  Avoiding that for the moment though to see what wiser minds know.

**Comment by mpenet** [comment]
Hi, 

I think it makes sense to support both if possible. 

I suppose we could adapt the specs to support both, that should be quite easy for the example you mentioned at least. https://github.com/mpenet/spandex/blob/master/src/clj/qbits/spandex/spec.clj#L129 `:req-un [(or ::response/host ::response/hosts)]`. Feel free to open a pr about these.

**Comment by jdt-at-litera** [comment]
Thanks for the reply.

Upon further digging it looks like the problem I encountered has nothing to do with Opensearch.  Our clojure shop uses [Orchestra](https://github.com/jeaye/orchestra) for clojure.spec instrumentation because unlike the default spec instrumentation it will validate the return codes.  I'm guessing the spandex contributors and/or other users aren't using
the return-validating instrumentation.

It turns out that Spandex' [Record](https://github.com/mpenet/spandex/blob/master/src/clj/qbits/spandex.clj#L199) definition has a field called 'hosts'.  This record is returned by the `request` function, but the spec for it is declaring `host` singular.  
Your `or` suggestion seems to work and doesn't make anything backward incompatible.

I added Orchestra to the profile.clj in my git clone and am trying to run your unit tests with it.  There were two other functions with incorrectly declared return types, easily fixed.  I'm still trying to figure out how you run the test.  If I just run `lein test` with an opensearch service running on port 9200 I'm still getting errors about "Connection is closed" for most tests, though it works more than it did without any Opensearch service running.
